cmd/kubeapps-apis/plugins/resources/v1alpha1/namespaces_filtering.go

```diff
@@ -40,44 +25,22 @@ func (s *Server) GetAccessibleNamespaces(ctx context.Context, cluster string, tr
 	if len(trustedNamespaces) > 0 {
 		namespaceList = append(namespaceList, trustedNamespaces...)
 	} else {
-
-		typedClient, err := s.clusterServiceAccountClientGetter.Typed(ctx, cluster)
-		if err != nil {
-			return nil, status.Errorf(codes.Internal, "unable to get the k8s client: '%v'", err)
+		clusterTypedClientFunc := func() (kubernetes.Interface, error) {
+			return s.clusterServiceAccountClientGetter.Typed(ctx, cluster)
+		}
+		inClusterTypedClientFunc := func() (kubernetes.Interface, error) {
+			return s.localServiceAccountClientGetter.Typed(context.Background())
```

Comment:
Just for the sake of knowledge, I don't understand why in this line you use `context.Background()` while above you're passing the existing `ctx` variable. Could you please explain the reason? 🙏🏻

Reply:
Most of the times, `clusterServiceAccountClientGetter` will use the context from the incoming request (`ctx` in this case), whereas `localServiceAccountClientGetter` does not need it, as it will provide a so called "in-cluster" client that will use the pod's service account. This local getter will have then different RBAC applied than the incoming context.